From c49e6c21df5869ead41c4a026bfd5f452ddaa3d4 Mon Sep 17 00:00:00 2001
From: NelloKudo <marshnelloosu@gmail.com>
Date: Wed, 10 Sep 2025 01:12:08 +0200
Subject: [PATCH] winediag: change logging and redirect to Spritz repository

---
 dlls/ntdll/loader.c | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/dlls/ntdll/loader.c b/dlls/ntdll/loader.c
index 1d3342f1c44..10ad0776165 100644
--- a/dlls/ntdll/loader.c
+++ b/dlls/ntdll/loader.c
@@ -4597,8 +4597,6 @@ void loader_init( CONTEXT *context, void **entry )
             if (FIXME_ON(sync))
             {
                 NTSTATUS sync = __wine_get_sync_type();
-                if (sync < 0)
-                    MESSAGE("ntsync is explicitly disabled.\n");
                 switch (abs(sync))
                 {
                 case 1:
@@ -4608,7 +4606,7 @@ void loader_init( CONTEXT *context, void **entry )
                     MESSAGE("fsync: up and running.\n");
                     break;
                 case 3:
-                    MESSAGE("wine: using fast synchronization.\n");
+                    MESSAGE("ntsync: up and running.\n");
                     break;
                 default:
                     MESSAGE("wineserver: using server-side synchronization.\n");
@@ -4616,7 +4614,10 @@ void loader_init( CONTEXT *context, void **entry )
                 }
             }
             if (FIXME_ON(winediag))
-                MESSAGE("winediag: this wine %s contains many experimental patches, please don't report bugs to winehq.org.\n", wine_get_version());
+            {
+                FIXME_(winediag)("Spritz-Wine-TkG (staging) %s is a downstream version containing specific patches for games.\n", wine_get_version());
+                FIXME_(winediag)("Please don't report bugs about it on winehq.org or at wine-tkg but use https://github.com/NelloKudo/Wine-Builds instead.\n");
+            }
         }
         NtClose( startup_event );
     }
-- 
2.51.0

