From 763098a64cd8fa59a5c9d2ce5bbc81233507c9e7 Mon Sep 17 00:00:00 2001
From: mkrsym1 <mkrsym1@gmail.com>
Date: Mon, 8 Sep 2025 21:05:27 +0200
Subject: [PATCH] loader: Copy arial.ttf to fonts

---
 loader/wine.inf.in | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/loader/wine.inf.in b/loader/wine.inf.in
index 892078a4a96..73b8cc0b9d7 100644
--- a/loader/wine.inf.in
+++ b/loader/wine.inf.in
@@ -54,7 +54,7 @@ AddService=MountMgr,0x800,MountMgrService
 RegisterDlls=RegisterDllsSection
 WineFakeDlls=FakeDlls
 UpdateInis=SystemIni
-CopyFiles=ColorFiles,EtcFiles,InfFiles,NlsFiles,SortFiles
+CopyFiles=ColorFiles,EtcFiles,InfFiles,NlsFiles,SortFiles,FontFiles
 AddReg=\
     Classes,\
     ContentIndex,\
@@ -111,7 +111,7 @@ AddReg=\
 [BaseWow64Install]
 RegisterDlls=RegisterDllsSection
 WineFakeDlls=FakeDllsWin32,FakeDllsWow64
-CopyFiles=NlsFiles
+CopyFiles=NlsFiles,FontFiles
 AddReg=\
     CurrentVersion,\
     Debugger,\
@@ -1295,9 +1295,13 @@ normnfkd.nls
 [SortFiles]
 sortdefault.nls
 
+[FontFiles]
+arial.ttf
+
 [WineSourceDirs]
 NlsFiles  = nls
 SortFiles = nls
+FontFiles = fonts
 
 [DestinationDirs]
 ColorFiles = 23
@@ -1305,3 +1309,4 @@ EtcFiles  = 12,etc
 InfFiles  = 17
 NlsFiles  = 11
 SortFiles = 10,globalization\sorting
+FontFiles = 20
-- 
2.49.1

